ktor-io/common/src/io/ktor/utils/io/CountedByteReadChannel.kt

```diff
@@ -36,15 +36,27 @@ public class CountedByteReadChannel(public val delegate: ByteReadChannel) : Byte
     @InternalAPI
     override val readBuffer: Buffer
         get() {
-            updateConsumed()
-            val appended = buffer.transferFrom(delegate.readBuffer)
-            initial += appended
+            transferFromDelegate()
             return buffer
         }
 
     @OptIn(InternalAPI::class)
     override suspend fun awaitContent(min: Int): Boolean {
-        return readBuffer.size >= min || delegate.awaitContent(min)
+        if (readBuffer.size >= min) {
+            return true
+        }
+        if (delegate.awaitContent(min)) {
+            transferFromDelegate()
```

Comment:
Is it okay that we transfer bytes from the buffer of the source channel in the `awaitContent` method?